Formula/elm-format.rb

```diff
@@ -15,27 +15,28 @@ class ElmFormat < Formula
   end
 
   depends_on "cabal-install" => :build
-  depends_on "ghc@8.8" => :build
+  depends_on "haskell-stack" => :build
+  depends_on arch: :x86_64 # no ghc (OSX,AArch64) binary via `haskell-stack`
 
-  def build_elm_format_conf
-    <<~EOS
-      module Build_elm_format where
+  uses_from_macos "xz" => :build # for `haskell-stack` to unpack ghc
 
-      gitDescribe :: String
-      gitDescribe = "#{version}"
-    EOS
+  on_linux do
+    depends_on "gmp" # for `haskell-stack` to configure ghc
```

Comment:
`stack setup` runs a test that uses `-lgmp`. Trying to add `gmp` as build-only dependency seems to cause `elm-format` to link to it. Not sure if we can avoid this somehow.  
```
==> brew linkage --test elm-format
==> FAILED
Unwanted system libraries:
  /usr/lib/x86_64-linux-gnu/libgmp.so.10
```

Reply:
Can you possibly pass `--with-intree-gmp` though `stack` to GHC when it's being built?  This would force the copy of GHC built by Stack to build its bundled copy of `gmp` and link to it statically, eliminating the need for `gmp` as a runtime dependency (it may still be needed for the bootstrap binary, which is only needed to build).

Reply:
It won't build a GHC; it just downloads a pre-built one.

Reply:
I guess we can't avoid this unless we build GHC inside `elm-format`, which would add quite the overhead.

The next step would probably be to check with upstream to see if they can loosen the dependency constraints so that we can build with our `ghc` formula.

There is probably some other dependency issues upstream given that both `cabal v2-install` fails and `stack init` fails

Reply:
I don't think a run time dependency on `libgmp` is a concern given that we're not actually building GHC.  When an upstream project has to resort to using a specific minor version of a compiler like this because they are unable or unwilling to keep up with GHC changes, extra runtime dependencies like this are a given.